### ISSUE TITLE
Add interactivity with SVG elements.

### ISSUE THREAD

**Issue by toolchild** [chrvadala/react-svg-pan-zoom#15]
Hi,

I'm trying to display a diagram with a lot of elements, which shall be interacted with.  The problem is that the group, in which the my original Diagram gets wrapped in has the `style="pointer-events: none;"`
So there is no mouse interaction with elements within that group.

**Example:**
```
const style = { pointerEvents: "all" };
<ReactSVGPanZoom ref={ref => this.elRefs.panZoomRoot = ref} width={this.props.width} height={this.props.height}
                         toolbarPosition="none"
                         className="diagram-pane-pan-zoom-root"
                         tool="pan"
                         style={style}
        >
```

Renders as:
```
<div style="position: relative; width: 712px; height: 832px;">
	<svg width="712" height="832" style="cursor: -webkit-grab; pointer-events: all;">
		<rect fill="#616264" x="0" y="0" width="712" height="832" style="pointer-events: none;"></rect>
		<g transform="matrix(0.4772809502278394, 0, 0, 0.4772809502278394, -1037.1809629679985, 64.83617277676811)" style="pointer-events: none;">
			<rect fill="#fff" x="0" y="0" width="10434" height="2451"></rect>
...
```

As you can see the group itself has a `style="pointer-events: none;"`. So there is no mouse interaction with anything inside that group.
It would be great if there was a way to control the styles, not only of the surrounding svg-element, but also of the inner elements created by the component.

regards,
toolchild

**Comment by chrvadala** [comment]
Hi,
the `pointer-events:none` applied to the wrapped elements is required by the tool pan that you have choose. With this tool enabled the component must have the full control of the events. If some child intercept the events used to perform pan(and zoom) operations, the component could have unexpected behaviour. If you want to interact with the child elements you have to active the tool: `none`.

What I could do is to introduce a new tool that work as you expect. This avoid to break the component and can solve a situation in which is required an interactivity with child tool. (I think that this could be very useful).

I will start this work very soon and I think that I need about one week. For now you have to use the toolbar or the feature autopan available out of the box.

Thank you for your feedback.

**Comment by chrvadala** [comment]
Hi @toolchild,
I've just released a new react-svg-pan-zoom version (2.2).
This new version introduce a new tool, called `auto` that work as you expected. Now you can interact with child elements, and perform pan/zoom operation without the need to switch to a different tool.

> With the tool auto the user can interact with SVG child elements, perform pan (dragging the image), zoom in (double click), zoom out (double click + shift).

Hope that this will be useful.
